src/index.ts

```diff
@@ -2534,25 +2534,31 @@ export default class Elysia<
 				plugin
 					.then((plugin) => {
 						if (typeof plugin === 'function') {
-							return plugin(
-								this as unknown as any
-							) as unknown as Elysia
+							return plugin(this)
+						}
+
+						if (plugin instanceof Elysia) {
+							return this._use(plugin)
 						}
 
-						if (typeof plugin.default === 'function')
-							return plugin.default(
-								this as unknown as any
-							) as unknown as Elysia
+						if (typeof plugin.default === 'function') {
+							return plugin.default(this)
+						}
 
-						return this._use(plugin as any)
```

Comment:
The issue was that `plugin.default` was not being referenced here, i.e. `plugin.default || plugin`. But also, I decided to clean up this area of the code, removing unnecessary type casts and adding `instanceof` checks for proper type narrowing.